### DISCUSSION TITLE
Nx Cloud Github App - Not seeing run results on PRs after setting up app

### DISCUSSION THREAD

**Discussion by parky128** [nrwl/nx#7500] [General]
I am attempting to integrate the Nx Cloud Github app with our github repository so we can get the run results appearing on our pull requests, but I am not seeing anything appear in my pull request comments\status checks area after following the necessary setup steps.

The repository is on public github, but does belong to an organisation - the repo has public visibility too.

The organisation admin has already added the Nx Cloud app from Github marketplace and made it available to all repo's on the org and when looking at the repo settings for where I'm trying to get this working, I see the app is listed:

![image](https://user-images.githubusercontent.com/10373419/138558023-69097bf0-8578-4297-b0bf-7c92f2df4285.png)

We are using Jenkins pipelines, and I have made sure to add the NX_BRANCH env variable, which I have setup like so:

```
pipeline {
    agent { label 'aws_slave_cypress' }
    environment {
        NX_BRANCH = env.BRANCH_NAME.replace('PR-', '')
    }
    stages {
        stage('Setup') {
            steps {
                sh 'npm ci'
            }
        }
        stage('Gate 2 - Build PR') {
            when {
                allOf {
                    branch "PR-*"
                }
            }
            steps {
                echo "GATE 2 - Build"
                sh "npx nx affected --base=origin/integration --target=build --parallel --max-parallel=3"
            }
        }
    }
}
```

When printing out our env variables in the pipeline, I see this is getting correctly set in the build output on Jenkins, e.g NX_BRANCH=4

We already have Nx Cloud configured, so have the correct accessToken setup in our nx.json

When my PR's build after following the Nx cloud setup steps, I still see the usual Nx Cloud run urls in the Jenkins output after the Nx task have been executed, but I dont see any Nx Cloud generated comments on my PR for the run results.

Also, if I go to our workspace settings on Nx Cloud, I can 'continue to setup' the workspace and when following the wizard steps and hitting Next button on the Open a Pull Request step, I see a notification telling me that no run report has been requested yet.

![image](https://user-images.githubusercontent.com/10373419/138558549-d1755b0b-119a-4a1c-8cbe-7d9ca09f439f.png)

So I am unsure what to do next here in terms of diagnosing why report runs are not appearing. I feel I have missed some sort of setup step, it maybe the Github app needs configuring further in our organisation settings.

Can anyone offer any clues as to what the problem could be here please?

Thanks

**Comment by junekpavel** [comment]
Hey! 

Have you figured out any solution? We're having same issue with TeamCity CI. 

Everything works fine, but the "Nx Cloud Report" comment isn't getting updated. The NX_BRANCH env variable is set to PR number. Are there any other requirements for this to work?

Thanks.

**Comment by stayko** [comment]
@parky128 I had a similar issue. I fixed it by adding a read only access token in `nx.json` and write and read token in CI. See here for more info: https://nx.dev/nx-cloud/account/scenarios
